packages/engine-server/src/drivers/sqlite/SqliteQueryUtils.ts

```diff
@@ -0,0 +1,154 @@
+import { DLogger } from "@dendronhq/common-all";
+import { getDurationMilliseconds } from "@dendronhq/common-server";
+import { ResultAsync } from "neverthrow";
+import { Database } from "sqlite3";
+import { SqliteError, SqliteErrorType } from "./SqliteError";
+
+/**
+ * Static Utils that help run sqlite database queries. Instead of taking
+ * callbacks like the sqlite3 library, these methods will return an awaitable
+ * ResultAsync and package Sqlite query errors into a SqliteError object.
+ */
+export class SqliteQueryUtils {
+  public static run(
+    db: Database,
+    sql: string,
+    logger?: DLogger
+  ): ResultAsync<null, SqliteError> {
+    const start = process.hrtime();
+
+    const prom = new Promise<null>((resolve, reject) => {
+      db.run(sql, (err) => {
```

Comment:
looks like this still silences errors.

I tried this:
```
  db.run(sql, (res: RunResult, err: Error) {
    ...
  }
```

running it on a same workspace that doesn't resolve to an `err` gives me a `res` that contains the actual error.

![image](https://user-images.githubusercontent.com/1219789/204779053-a9cb6eaa-252b-4bc7-b604-cc6641b11fc0.png)

Reply:
```typescript
  db.run(sql, (res: RunResult, err: Error) {
    ...
  }
```

db.run's callback only has one argument, which is err.  So if you put it as

```typescript
  db.run(sql, (err: Error) {
    ...
  }
```
then it should come back as `err`